src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    static final String DELIMITER = ",";
+    static final String OPPERATION_SUPPLY = "supply";
+    static final String OPPERATION_BUY = "buy";
+    static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))
```

Comment:
Please check common mistakes
https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#divide-program-logic
Create 3 private methods (read file, prepare report and write report to file), split logic properly between them and in you public method just call these 3 methods